rust/lance-core/src/format/manifest.rs

```diff
@@ -389,6 +394,50 @@ impl From<&Manifest> for pb::Manifest {
     }
 }
 
+#[async_trait]
+pub trait SelfDescribingFileReader {
+    /// Open a file reader without any cached schema
+    ///
+    /// In this case the schema will first need to be loaded
+    /// from the file itself.
+    ///
+    /// When loading files from a dataset it is preferable to use
+    /// the fragment reader to avoid this overhead.
+    async fn try_new_self_described(
```

Comment:
I made this an extension function because I expect this method to move back into the file reader once we move toward a new version of the metadata that stores a schema (and not a manifest) in the file.